src/nethsec/dpi/__init__.py

```diff
@@ -0,0 +1,302 @@
+        result.append(result_application)
+
+    protocols = __load_protocols()
+    protocol_categories = __load_protocol_categories()
+
+    for protocol_id, protocol_name in protocols.items():
+        result_protocol = {
+            'id': protocol_id,
+            'name': protocol_name,
+            'type': 'protocol'
+        }
+        if protocol_id in protocol_categories:
+            result_protocol['category'] = protocol_categories[protocol_id]
+        result.append(result_protocol)
+
+    return result
+
+
+def list_applications(search: str = None, limit: int = None, page: int = 1) -> list[dict[str, str]]:
+    """
+    List applications available for filtering.
+
+    Args:
+      - search: search string
+      - limit: limit the number of results
+      - page: page number
+
+    Returns:
+        list of dicts, each dict contains the property "code" and "name"
+    """
+    result = __load_blocklist()
+
+    if search is not None:
+        # lower string so we can do a case-insensitive search
+        search = search.lower()
+        # I'm aware it's far from a readable code, but list comprehension is the fastest way to filter.
+        result = [item for item in result if
+                  item.get('name', '').lower().startswith(search) or
+                  item.get('category', {}).get('name', '').lower().startswith(search)]
+
+    if limit is not None:
+        result = result[limit * (page - 1):limit * page]
+
+    return result
+
+
+def list_rules(e_uci: EUci) -> list[dict[str]]:
+    """
+    Index all rules
+
+    Args:
+      - e_uci: euci instance
+
+    Returns:
+        list of dicts, each dict contains the property "config-name", "description", "enabled", "interface" and "blocks"
+    """
+    rules = list[dict[str]]()
+    fetch_rules = utils.get_all_by_type(e_uci, 'dpi', 'rule')
+
+    if not fetch_rules:
+        return rules
+
+    for rule_name in fetch_rules.keys():
+        # skipping rules with criteria, must be custom entries
+        if e_uci.get('dpi', rule_name, 'criteria', default=None) is None:
+            # load blocklist of applications and protocols
+            blocklist = __load_blocklist()
+            # get content of rule
+            rule = fetch_rules[rule_name]
+            # prepare the data to append to rules
+            data_rule = dict[str]()
+            data_rule['config-name'] = rule_name
+            data_rule['enabled'] = rule.get('enabled', '1') == '1'
+            data_rule['interface'] = rule.get('interface', '*')
+            # get the blocked applications/protocols
+            data_rule['blocks'] = list[dict[str]]()
+
+            # filter by application
+            application_blocklist = [item for item in blocklist if item['type'] == 'application']
+            for application in rule.get('application', []):
+                found_app = [item for item in application_blocklist if
+                             item['name'] == application.removeprefix('netify.')]
```

Comment:
Do not use 'netify.' as a fixed prefix.

Reply:
Same as above, returning full identifier of application rule, letting the UI handle this split